### ISSUE TITLE
Gradient Accumulation Support

### ISSUE THREAD

**Issue by mmderakhshani** [KaiyangZhou/Dassl.pytorch#54]
Hi @KaiyangZhou, thanks for implementing this excellent framework. I was wondering if there is any way to add the functionality for gradient accumulation.

**Comment by KaiyangZhou** [comment]
yes, instead of using `self.model_backward_and_update()`, you can implement the backward steps yourself (in which you do gradient accumulation)

**Comment by mmderakhshani** [comment]
Thanks!
